### PR TITLE
Corrected Cleanup, Added Verbose Option

### PR DESCRIPTION
**Pull request by louis-tk**
Moved file closure and clean-up to occurr after the output stream in 'union' to prevent errors in the output.
Added 'verbose' option, defaulted to False to prevent 'print' commands when being run as a module.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+
+*.xml
```
Comment (tkmru) [inline .gitignore]: why do you add xml?

**Review by tkmru — CHANGES_REQUESTED** [review]

**Comment by tkmru** [comment]
Your pull request work correctly. Awesome. Please fix gitignore.

File: .gitignore (hunk 1 above)
Comment (louis-tk) [reply .gitignore]: This has been added so that extra files created by the PyCharm editor are ignored by Git.

Comment (tkmru) [reply .gitignore]: Please delete it, because xml is not unrelated module.

**Comment by tkmru** [comment]
Thanks!!

**Pull request merged**